### PR TITLE
Pollset

### PR DESCRIPTION
**Pull request by hnwyllmm**
add a new poller named pollset which will get benefit of performance in AIX platform.

### CODE REVIEW

**Comment by somdoron** [comment]
@hnwyllmm you broke the build, can you move the new cpp and hpp files to the src directory?

File: CMakeLists.txt [hunk 1]
```diff
@@ -92,6 +92,15 @@ if (POLLER STREQUAL "")
 endif ()
 
 if (POLLER STREQUAL "")
+    set (CMAKE_REQUIRED_INCLUDES sys/pollset.h)
```
Comment (bluca) [inline CMakeLists.txt]: Please don't change the default, move this snippet below

**Comment by bluca** [comment]
Thanks for sending the pr!

Apart from the comments from Doron and myself, a couple suggestions. If you have multiple commits, use a different commit message for each one. Also each commit should work by itself (by which I mean not break the build and the tests) if possible.

Also, for your next contributions, could you please follow the contributing guidelines for the git commit message?

```
Problem: X is broken

Solution: do Y and Z to fix X
```

http://zeromq.org/docs:contributing

File: CMakeLists.txt (hunk 1 above)
Comment (hnwyllmm) [inline CMakeLists.txt]: I will move it just before `poll` just like below, does it ok for that? (I am so sorry for my stupid commit)
`if (POLLER STREQUAL "")
    set (CMAKE_REQUIRED_INCLUDES sys/devpoll.h)
    check_type_size ("struct pollfd" DEVPOLL)
    set (CMAKE_REQUIRED_INCLUDES)
    if (HAVE_DEVPOLL)
        set (POLLER "devpoll")
    endif ()
endif ()

if (POLLER STREQUAL "")
    set (CMAKE_REQUIRED_INCLUDES sys/pollset.h)
    check_function_exists (pollset_create HAVE_POLLSET)
    set (CMAKE_REQUIRED_INCLUDES)
    if (HAVE_POLLSET)
        set (POLLER "pollset")
    endif()
endif ()

if (POLLER STREQUAL "")
    set (CMAKE_REQUIRED_INCLUDES poll.h)
    check_function_exists (poll HAVE_POLL)
    set (CMAKE_REQUIRED_INCLUDES)
    if (HAVE_POLL)
        set (POLLER "poll")
    endif ()
endif ()`

Comment (bluca) [inline CMakeLists.txt]: Yes that would be perfect, thank you!

File: CMakeLists.txt [hunk 2]
```diff
@@ -468,6 +478,7 @@ set (cxx-sources
         plain_server.cpp
         poll.cpp
         poller_base.cpp
+		pollset.cpp
```
Comment (bluca) [inline CMakeLists.txt]: In the next commit could you also please fix the indentation? It's spaces only, no tabs

File: acinclude.m4 [hunk 3]
```diff
@@ -901,7 +917,7 @@ AC_DEFUN([LIBZMQ_CHECK_POLLER], [{
     fi
     if test "$pollers" == "auto"; then
         # We search for pollers in this order
-        pollers="kqueue epoll devpoll poll select"
+        pollers="kqueue pollset epoll devpoll poll select"
```
Comment (bluca) [inline acinclude.m4]: Please also move this after devpoll, we do not want to change defaults

File: CMakeLists.txt (hunk 2 above)
Comment (hnwyllmm) [inline CMakeLists.txt]: Done, please view #2120 , Thanks

File: acinclude.m4 (hunk 3 above)
Comment (hnwyllmm) [inline acinclude.m4]: Done. Please see #2120

**Pull request merged**